Sources/ComposableArchitecture/Reducer/ReducerBuilder.swift

```diff
@@ -385,18 +385,18 @@ public enum ReducerBuilder<State, Action> {
     First.State == Second.State,
     First.Action == Second.Action
   {
-    case first(First)
-    case second(Second)
+    case first(First, Second.Type = Second.self)
+    case second(Second, First.Type = First.self)
```

Comment:
I hadn't thought to include the type in the case. Did you try a static func as well?

```swift
@usableFromInline
static func first(_ reducer: First, _ secondType: Second.Type) -> Self {
  .first(reducer)
}
```

Is there a benefit including it as an associated value?

Reply:
@iampatbrown I spoke too soon, using it as either associated value or a static func doesn't appear to compile, unfortunately. Definitely down to merge something that compiles, but both my commit and your code above crashes the compiler 😞

Reply:
@stephencelis So close.. yet so far...
`buildEither` probably wants a concrete type in order to match `first:` and `second:`.
I think this PR is great without it anyway.